source/extensions/health_checkers/tcp/health_checker_impl.cc

```diff
@@ -141,12 +146,21 @@ void TcpHealthCheckerImpl::TcpActiveHealthCheckSession::onInterval() {
     client_->noDelay(true);
   }
 
+  Buffer::OwnedImpl data;
+  bool should_write_data = false;
+
+  if (parent_.proxy_protocol_config_ != nullptr) {
+    Extensions::Common::ProxyProtocol::generateProxyProtoHeader(*parent_.proxy_protocol_config_,
```

Comment:
There another proxy protocol command "local" which looks like it's meant for health checks (and that the remote server should use the connection endpoints and causes the endpoints in the protocol to be ignored). I *think* it's okay to use the command this function generates since we'll set the endpoints to be ourselves the server being health checked, but I thought I'd point it out.

Reply:
Yes. Actually we can use local command for filling the requirements to check whether ProxyProtocol is enabled by healthcheck. But I think that we can configure more fine-grained healthcheck strategy by setting TLVs defined from ProxyProtocol config. So I think It's better to use `generateProxyProtocolHeader` for advanced requirements.

Reply:
Could you elaborate more for the meaning of address setting in the Proxy Protocol header? The health check's local address is the upstream local address, and the remote is the health check address of the remote host. I think this is different from the real upstream request which is using the downstream connection as the input of this `generateProxyProtoHeader` util.

I'd incline to add some comments for this, and describe the `LOCAL` as well as the address setting. Also could you add a test to see if the proxy protocol with address is actually sent/received?

Reply:
@botengyao @zuercher Hi. Sorry for late response.
I revisited how to provide this feature and came up with that we should provide this feature with local command for V2, and present local/remote address for V1 because it doesn't support them. This is why I found that some users faced confusion for this behavior to provide with PROXY command for healthcheck in HAProxy community. 
 https://github.com/haproxy/haproxy/issues/511

So, I cleaned entire codes in this patch to support them.